File: w2/db_query/backend/app/models/query.py
from typing import Any, List, Optional
from pydantic import BaseModel, Field


class QueryRequest(BaseModel):
    connection_id: int = Field(..., alias="connectionId")
    sql: str

    class Config:
        populate_by_name = True


class QueryColumn(BaseModel):
    name: str
    type: Optional[str] = None


class QueryResult(BaseModel):
    columns: List[QueryColumn]
    rows: List[List[Any]]
    limit_added: bool = Field(False, alias="limitAdded")
    message: Optional[str] = None

    class Config:
        populate_by_name = True

File: w2/db_query/frontend/src/services/query.ts
import { apiClient } from './api'

export interface RunQueryPayload {
  connectionId: number
  sql: string
}

export interface QueryResult {
  columns: { name: string; type?: string }[]
  rows: any[][]
  limitAdded: boolean
  message?: string
}

export async function runQuery(payload: RunQueryPayload): Promise<QueryResult> {
  const { data } = await apiClient.post('/query', payload)
  return data
}

